Adding a couple turrets-hosting-related docs

ElliotFriend opened the pull request:
Copied the reference implementation, and wrote a high-level overview of what a turret host is actually hosting.

ElliotFriend committed:
Porting the whole 'Reference Implementation' over from the README

Signed-off-by: Elliot J. Voris <elliot@voris.me>

ElliotFriend committed:
Forgot to fix some stuff from the cloudToButt extension... Oops

Signed-off-by: Elliot J. Voris <elliot@voris.me>

ElliotFriend committed:
Created `anatomy-of-a-turret.md`

Signed-off-by: Elliot J. Voris <elliot@voris.me>

mootz12 commented on turret-hosts/anatomy-of-a-turret.md:
```diff
@@ -0,0 +1,58 @@
+# Anatomy of a Turret
```
This was a good addition!

ElliotFriend replied:
Thanks!

mootz12 commented on turret-hosts/reference-implementation.md:
```diff
@@ -0,0 +1,216 @@
+# Stellar Turrets Reference Implementation
```
Nice!

A cool addition would be some of the details in https://github.com/tyvdh/stellar-turrets/pull/36 about running more than one turret

ElliotFriend replied:
Good call! Maybe I'll add some notes about testnet vs mainnet with a link to Tyler's PR, since it's still a work-in-progress?

mootz12 reviewed (commented):
Looks good! 

See one comment about additional material for turret hosts. This is NOT a requirement for approval, just an additional suggestion!

ElliotFriend committed:
PR #2: Adding a bit of a note pointing towards the mainnet info

Added a description of running two turrets, and linked to
tyvdh/stellar-turrets#36 for further information.

Signed-off-by: Elliot J. Voris <elliot@voris.me>
